qiskit/pulse/transforms.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Basic rescheduling functions which take schedules or instructions
+(and possibly some arguments) and return new schedules.
+"""
+from collections import defaultdict
+from typing import List
+
+from qiskit.pulse import Play, Schedule
+
+
+def compress_pulses(schedules: List[Schedule],
+                    by_schedule: bool = False,
+                    by_channel: bool = False) -> List[Schedule]:
+    """Optimization pass to replace identical pulses.
+
+    Args:
+        schedules (list): Schedules to compress.
+        by_schedule (bool): Compress pulses by schedule.
+        by_channel (bool): Compress pulses by channel.
```

Comment:
What is the use case for these?

Reply:
Just to provide more granularity on applied operation. I can remove channel if needed

Reply:
I can't think of use case for reducing the effectiveness of the function by channel, so let's remove it. I'm not sure I see the use case for `by_schedule` either